From e0fb99d14c3ed1140495f179090a3504791a9103 Mon Sep 17 00:00:00 2001
From: Sidhanta Sahu <quic_sidhanta@quicinc.com>
Date: Thu, 28 Dec 2023 10:52:30 -0800
Subject: [PATCH] PPEDS - Add mlnetdev variable for handling bonded
dev creation

Add support to handle the vendor command notifying the presence
of an MLD before hand, during link addition, which will help
in destroying bond netdev for SLD interfaces

Signed-off-by: Balamurugan Mahalingam <quic_bmahalin@quicinc.com>
Signed-off-by: Ramanathan Choodamani <quic_rchoodam@quicinc.com>
Signed-off-by: Sidhanta Sahu <quic_sidhanta@quicinc.com>
---
 include/net/cfg80211.h | 1 +
 include/net/mac80211.h | 1 +
 2 files changed, 2 insertions(+)

--- a/include/net/cfg80211.h
+++ b/include/net/cfg80211.h
@@ -6285,6 +6285,7 @@ struct wireless_dev {
 	struct list_head list;
 	struct net_device *netdev;
 	struct net_device *bond_netdev;
+	bool ml_netdev;
 
 	u32 identifier;
 
--- a/include/net/mac80211.h
+++ b/include/net/mac80211.h
@@ -1972,6 +1972,7 @@ struct ieee80211_vif {
 	struct ieee80211_bss_conf bss_conf;
 	struct ieee80211_bss_conf __rcu *link_conf[IEEE80211_MLD_MAX_NUM_LINKS];
 	u16 valid_links, active_links, dormant_links;
+	bool ml_netdev;
 	u8 addr[ETH_ALEN] __aligned(2);
 	bool p2p;
 
